src/fifelse.c

```diff
@@ -257,13 +257,13 @@ SEXP fcaseR(SEXP na, SEXP rho, SEXP args) {
     case INTSXP: {
       const int *restrict pouts = INTEGER(outs);
       int *restrict pans = INTEGER(ans);
-      const int pna = nonna ? INTEGER(na)[0] : NA_INTEGER;
       for (int64_t j=0; j<len2; ++j) {
         idx = imask ? j : p[j];
         if (pcons[idx]==1) {
           pans[idx] = pouts[idx & amask];
         } else {
           if (imask) {
+            const int pna = nonna ? INTEGER(na)[scalarna?0:j] : NA_INTEGER;
```

Comment:
we should avoid repeatedly calling `INTEGER` inside the loop by assigning the `int *` pointer to `INTEGER(na)` before the loop starts